src/components/FileTreeNode/index.tsx

```diff
@@ -50,6 +50,24 @@ export const findMostSevereTypeForPath = (
   return null;
 };
 
+export const LINTER_KNOWN_LIBRARY_CODE = 'KNOWN_LIBRARY';
+
+export const isKnownLibrary = (
+  linterMessageMap: LinterMessageMap,
+  path: string,
+): boolean => {
+  if (!linterMessageMap[path]) {
+    return false;
+  }
+
+  const m = linterMessageMap[path];
+  if (Object.keys(m.global).length > 1 || Object.keys(m.byLine).length > 0) {
```

Comment:
I would like to see this account for future keys that we might add to `linterMessageMap`. You could do it a couple ways. Ideas:

- Assume that all keys on `linterMessageMap` will be arrays and concat them all together
- Iterate through a list of known keys (`global`, `byLine`) and throw an error if a new key is present

It's a bit paranoid but, as it stands, the known library code would win against messages in a new mapping key since there are no failing tests or type checks to remind us to update this part of the code. For something as sensitive as hiding files, I want to be sure that this will do the right thing.

Reply:
I see, thanks!

Reply:
> Assume that all keys on `linterMessageMap` will be arrays and concat them all together

that won't work because `byLine` is a map too.

Reply:
I went for the error thrown if a key is unexpected.